### ISSUE TITLE
Pyright does not show error when mixin's attribute was not initialized

### ISSUE THREAD

**Issue by AlexandrKrivosheev** [microsoft/pyright#1087]
**Describe the bug**
Pyright does not show error when mixin's attribute was not initialized

**Expected behavior**
Pyright should show that initialization of attribute `a_param` missing in object `AB`

**Screenshots or Code**
```python
class AMixin:
    a_param: str

    def a(self):
        return self.a_param


class AB(AMixin):
    def __init__(self, b: str) -> None:
        self.b_param = b

    def b(self):
        return self.b_param


ab = AB("b")
ab.a() # will throw because `AB` object has no attribute `a_param`
```

**VS Code extension or command-line**
vscode pylance v2020.9.8

**Comment by erictraut** [comment]
There are limits to code errors that a static type checker can find. I don't think this is a case that a static type checker can feasibly detect. Mypy doesn't report any problem here either.

From the type checker's perspective, you have declared that `AMixin` has a class variable whose type is `str`. There are many ways in which such a variable could be initialized. 

For example:
```python
ab = AB("b")
fill_in_a_param(ab) # Fills in the value of `a_param` within AB instance
ab.a()
```

Pyright is able to detect and report unbound variables within the same execution scope. For example:

```python
def foo():
    a: str
    b = a # "a" is unbound
```

This is possible through the use of a control-flow graph within the function's execution scope. However, in the example above, there are many execution scopes.

**Comment by AlexandrKrivosheev** [comment]
in case of `fill_in_a_param` `a_param` has to be `Optional[str]`, but if `a_param` is not optional, it has to be assigned in `__init__`, no?

**Comment by erictraut** [comment]
No, a class variable type annotation does not imply that the variable must be bound to a value within the `__init__` method. And `Optional` does not have any special interpretation with regard to binding. `Optional` simply means that you are allowed to bind the variable with a value that is either type `None` or another type.

**Comment by AlexandrKrivosheev** [comment]
I see, thank you 🙏 😁
